rpc/rest/security/rest_security.go

```diff
@@ -0,0 +1,415 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate protoc --proto_path=model/http-security --gogo_out=model/http-security model/http-security/httpsecurity.proto
+
+package security
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/gorilla/mux"
+	"github.com/ligato/cn-infra/logging"
+	httpsecurity "github.com/ligato/cn-infra/rpc/rest/security/model/http-security"
+	"github.com/pkg/errors"
+	"github.com/unrolled/render"
+	"golang.org/x/crypto/bcrypt"
+)
+
+const (
+	// Helps to obtain authorization header matching the field in a request
+	authLabel = "authorization"
+	// Admin constant, used to define admin security group and user
+	admin = "admin"
+)
+
+const (
+	// Returns login page where credentials may be put. Redirects to authenticate, and if successful, moves to index.
+	login = "/login"
+	// URL key for logout, invalidates current token.
+	logout = "/logout"
+	// Authentication page, validates credentials and if successful, returns a token or writes a cookie to a browser
+	authenticate = "/authenticate"
+	// Cookie name identifier
+	cookieName = "ligato-rest"
+)
+
+// Default value to sign the token, if not provided from config file
+var signature = "secret"
+
+// Default expiration time for token/cookie
+var expTime time.Duration = 3600000000000 // 1 Hour
+
+// AuthenticatorAPI provides methods for handling permissions
+type AuthenticatorAPI interface {
+	// AddPermissionGroup adds new permission group. PG is defined by name and a set of URL keys. User with
+	// permission group enabled has access to that set of keys. PGs with duplicated names are skipped.
+	AddPermissionGroup(group ...*httpsecurity.PermissionGroup)
+
+	// Validate serves as middleware used while registering new HTTP handler. For every request, token
+	// and permission group is validated.
+	Validate(provider http.HandlerFunc) http.HandlerFunc
+}
+
+// Context defines fields required to instantiate authenticator
+type Context struct {
```

Comment:
The word **context** and also use of such package is generally used for request scoped (time based) contexts, which this isn't at all. This is either *options*, *config* or *settings*, since those represents something persistent.